src/Command/Argument/Search/Schema.php

```diff
@@ -0,0 +1,191 @@
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addTextField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('TEXT', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds tag field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addTagField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('TAG', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds numeric field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addNumericField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('NUMERIC', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds geo field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addGeoField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('GEO', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds vector field to schema configuration.
+     *
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    public function addVectorField(
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
+        bool $unf = false,
+        bool $noIndex = false
+    ): self {
+        return $this->addField('VECTOR', $fieldName, $alias, $sortable, $unf, $noIndex);
+    }
+
+    /**
+     * Adds given field to schema configuration.
+     *
+     * @param  string $type
+     * @param  string $fieldName
+     * @param  string $alias
+     * @param  bool   $sortable
+     * @param  bool   $unf
+     * @param  bool   $noIndex
+     * @return $this
+     */
+    private function addField(
+        string $type,
+        string $fieldName,
+        string $alias = '',
+        bool $sortable = false,
```

Comment:
1. Not all fields are `SORTABLE`, only `NUMERIC`, `TAG`, `TEXT` and `GEO` are.
2. `$sortable = false, $unf = true` is invalid... I think you should use an `enum` (or any equivalent alternative):
```php
enum Sortable
{
    case False; // rename to `Off`/`False`?
    case True; // rename to `On`/`True`?
    case UNF;
}

private function addField(
    string $type,
    string $fieldName,
    string $alias = '',
    bool $sortable = Sortable.False,
    bool $noIndex = false
)
```

Reply:
@leibale 
1. Okay, thanks for letting me know 👌
2. Enums in PHP supported only from >= 8.1, but in Predis we should support >= 7.2

Reply:
Fixed

Reply:
@vladvildanov you can use constants instead of an enum

Reply:
@leibale Fixed